### PR TITLE
Requirement for voter reward - component tests

### PR DESCRIPTION
**Pull request by nicopado**
Corrected component level tests for voter rewards requirement, updated gitignore to exclude ide-specific folders (vs and idea)

### CODE REVIEW

File: catalyst-toolbox/src/rewards/voters.rs [hunk 1]
```diff
@@ -247,27 +283,29 @@ mod tests {
...
             .iter()
-            .all(|addr| rewards.remove(addr).unwrap() > Rewards::ZERO));
+            .all(|addr| all_rewards.remove(addr).unwrap() > Rewards::ZERO));
 
         // partial test: does not check that rewards for addresses that delegated to both
-        // active and inactive voters only come from active ones
-        for (_, vk) in inactive {
-            for contrib in snapshot.contributions_for_voting_key(vk.clone()) {
-                assert!(rewards.get(&contrib.reward_address).is_none());
+        // active and inactive voters only come from active ones -> how to implement this?
```
Comment (kukkok3) [inline catalyst-toolbox/src/rewards/voters.rs:289]: Leftover comment?

**Review by kukkok3 — APPROVED** [review]
LGTM Just check and fix the two leftover comments if it is needed

File: catalyst-toolbox/src/rewards/voters.rs [hunk 2]
```diff
@@ -136,49 +136,80 @@ mod tests {
...
             votes_count,
             voters,
-            Threshold::new(1, HashMap::new(), Vec::new()).unwrap(),
+            Threshold::new(DEFAULT_TEST_THRESHOLD, HashMap::new(), Vec::new()).unwrap(),
             Rewards::ONE,
         )
         .unwrap();
-        if n_voters > 0 {
+        if number_of_voters > 0 {
+            // number of voters always 0 ???
```
[the remark this answers is not in the document]
Comment (nicopado) [reply catalyst-toolbox/src/rewards/voters.rs:173]: yap, good catch!

File: catalyst-toolbox/src/rewards/voters.rs (hunk 1 above)
Comment (nicopado) [reply catalyst-toolbox/src/rewards/voters.rs:289]: yas, still me forgetting comments around

**Pull request merged**